agents/xenapi/fence_xenapi.py

```diff
@@ -191,23 +196,42 @@ def return_vm_reference(session, options):
 	# to the VM).
 	raise Exception("VM_LOGIC_ERROR")
 
+def define_new_opts():
+	all_opt["uuid"] = {
+		"getopt" : ":",
+		"longopt" : "uuid",
+		"help" : "--uuid=[VM UUID]              The VM UUID.",
+		"default" : "",
+		"required" : "0",
+		"shortdesc" : "The VM UUID",
+		"order" : 2}
+	all_opt["plug-separator"] = {
+		"getopt" : ":",
+		"longopt" : "plug-separator",
+		"help" : "--plug-separator=separator              The saparator for multiple plugs.",
+		"default" : ":",
+		"required" : "0",
+		"shortdesc" : "The saparator for multiple plugs",
+		"order" : 3}
+
 def main():
 
-	device_opt = ["login", "passwd", "port", "no_login", "no_password", "session_url", "web"]
+	device_opt = ["login", "passwd", "port", "no_login", "no_password", "session_url", "web", "ssl", "uuid", "plug_separator"]
```

Comment:
plug_separator is included in "port" from fencing.py, so you can remove that and it's definition above.

uuid isnt neither either, as the fencing library sets it based on the --plug/-n parameter.

Reply:
I couldn't get it to work without these two parameters. Maybe I'm just testing it wrong? This way works for me and I have no way of validating other solutions, so someone else might need to take over this pull request (or just delete it if you prefer).

Reply:
You probably need to update your systems fencing library (usually located in /usr/share/fence).